WalletWasabi/BitcoinCore/Mempool/MempoolMirror.cs

```diff
@@ -102,9 +103,10 @@ private async Task<int> MirrorMempoolAsync(CancellationToken cancel)
 		IEnumerable<uint256> missing;
 		lock (MempoolLock)
 		{
-			missing = mempoolHashes.Except(Mempool.Keys);
+			var mempoolKeys = Mempool.Keys.ToImmutableArray();
+			missing = mempoolHashes.Except(mempoolKeys);
 
-			foreach (var txid in Mempool.Keys.Except(mempoolHashes).ToHashSet())
+			foreach (var txid in mempoolKeys.Except(mempoolHashes).ToHashSet())
```

Comment:
I suspect that `.ToHashSet()` may be extra as `Except` probably creates new immutable array

Reply:
I think the error has happened because `missing` was changed. This PR fixes it as `ToImmutableArray()` clones the array. `ToHashSet()` became useless.